### ISSUE TITLE
Turning on PSMove controller when PSMoveService/Steam VR running causes SteamVR to shutdown

### ISSUE THREAD

**Issue by ToppestOfDogs** [cboulay/PSMoveService#238]
With PSMoveService running, the controllers will not turn on, the cameras are not detected, the headset will not respond to movement, and Steam VR cannot close. Without PSMoveService running it works fine.
I have PSMoveService all set up and tracking properly, so the controllers not turning on is not because of them not being set up.
I initially thought this was a problem with 5.3, but it is also happening with 5.1 which worked previously, so it may be because of a change with Steam.
I've tried restarting the computer and reinstalling the driver but the problem persists.
Running with two cameras, two controllers, and RiftCat (Which works fine with PSMoveService not running)

**Comment by HipsterSloth** [comment]
Hmm.. Maybe the steamvr client is getting stuck trying to connect to psmoveservice? I assume when you said "The controllers will not turn on, the cameras are not detected..." you meant inside the SteamVR monitor window when you launched SteamVR?

 Some troubleshooting tasks for you:

1) Can you get camera feeds if you run test_cameras.exe without psmove service running?
2) Does the "Test tracking" tool in PSMoveConfigTool track the controllers ok?
3) Do you get an error code in the SteamVR windows? Like error 301?
3) Can you clear out your `Program Files (x86)\Steam\logs`, launch steamvr and wait until it hangs, then save a copy of the vrserver.txt file that gets generated here. I;m hoping some helpful error shows up in the logs

**Comment by ToppestOfDogs** [comment]
When I say the controllers will not turn on, I mean the lights do not turn on.
The camera feeds work, and test tracking works. Everything in the config tool seems to be working fine as far as I can tell. There is the error "libusb: error [windows_get_device_list] program assertion failed - existing device should share parent" in the console but everything is working fine.
Unfortunately there is no error in steam, all that happens is when the driver is started (And I've tried starting before and after launching Steam VR) the headset stops moving, and when I try to close steam vr it just stays and I have to close everything from the task manager.
[vrserver.txt](https://github.com/cboulay/PSMoveService/files/520974/vrserver.txt)

**Comment by HipsterSloth** [comment]
> When I say the controllers will not turn on, I mean the lights do not turn on.

This can mean a couple of things, but the most like issue is that the steamvr client hasn't told the controllers to start tracking yet.

> There is the error "libusb: error [windows_get_device_list] program assertion failed - existing device should share parent" in the console but everything is working fine.

I see that error all the time. It's because I have a debug log setting checked with libusb. It doesn't actually harm anything.

>  when I try to close steam vr it just stays and I have to close everything from the task manager.

I think this is a zombie vrserver.exe issue. From your vrserver log it looks like it's connecting to the PSMove controllers:

```
Mon Oct 10 2016 19:36:54.134 - psmove: NotifyClientPSMoveResponse - Controller Count = 2 (request id 0).
Mon Oct 10 2016 19:36:54.134 - psmove: added new psmove controller id: psmove_controller0, serial: 00:06:F5:85:B6:1A
...
Mon Oct 10 2016 19:36:54.134 - psmove: added new psmove controller id: psmove_controller1, serial: 00:06:F5:17:FA:60
```

But then vrmonitor is shutting down vrserver:

```
Mon Oct 10 2016 19:37:06.620 - Process vrmonitor (9992) has initiated a quit all
Mon Oct 10 2016 19:37:15.195 - IPCPipe(vrdashboard (11384)): Process disconnected.
Mon Oct 10 2016 19:37:15.195 - Process vrdashboard (11384) disconnected
```

I think this is the same problem as issue #239 where there is a zombie vrserver.exe still running after closing steamvr. Try shutting down steamvr and psmoveserver. Then look in process monitor for any remaining instances of vrserver.exe. Kill those. Then restart psmoveservice and steamvr and see if that fixes the problem. 

If the problem still happens could you attach the latest vrserver.txt and vrmonitor.txt logs?

**Comment by HipsterSloth** [comment]
I'm going to use issue #239 to tracking this problem (there were three separate issue posted for what I believe is the same underlying steamvr startup problem). I'm going to close this issue for tracking purposes (just to try and keep the open issues list clean), but please keep posting to this thread as we debug this.

**Comment by ToppestOfDogs** [comment]
I've been shutting down the vrserver.exe and everything else related to steam vr each time, so that isn't the issue. I just tried again with the cameras plugged in and the controller turned off, and steam could detect the cameras and worked fine, the models showed up in the vr space like they should, but as soon as I turned the controller on by pressing the PS button, it stopped working and did the same thing as before.
[vrserver.txt](https://github.com/cboulay/PSMoveService/files/522635/vrserver.txt)

**Comment by HipsterSloth** [comment]
Same story in that log as before. Controller connect but vrmonitor closes vrserver:

```
Tue Oct 11 2016 14:31:23.399 - psmove: NotifyClientPSMoveResponse - Controller Count = 1 (request id 2).
Tue Oct 11 2016 14:31:23.399 - psmove: added new psmove controller id: psmove_controller0, serial: 00:06:F5:17:FA:60
Tue Oct 11 2016 14:31:23.399 - psmove: m_fControllerMetersInFrontOfHmdAtCallibration: -536870912
Tue Oct 11 2016 14:31:23.410 - psmove: CPSMoveTrackedDeviceLatest::Activate: psmove_controller0 is object id 3
Tue Oct 11 2016 14:31:23.410 - psmove: CPSMoveControllerLatest::Activate(3) -- calling g_ServerTrackedDeviceProvider.LaunchPSMoveMonitor()
Tue Oct 11 2016 14:31:23.410 - psmove: Entered CServerDriver_PSMoveService::LaunchPSMoveMonitor()
Tue Oct 11 2016 14:31:23.410 - psmove: Entered CServerDriver_PSMoveService::LaunchPSMoveMonitor(C:\Program Files (x86)\Steam\steamapps\common\SteamVR\drivers\psmove)
Tue Oct 11 2016 14:31:23.410 - psmove: CServerDriver_PSMoveService::LaunchPSMoveMonitor() monitor_psmove windows full path: "C:\Program Files (x86)\Steam\steamapps\common\SteamVR\drivers\psmove\bin\win64\monitor_psmove.exe"
Tue Oct 11 2016 14:31:23.410 - psmove: CServerDriver_PSMoveService::LaunchPSMoveMonitor() monitor_psmove windows args: "C:\Program Files (x86)\Steam\steamapps\common\SteamVR\drivers\psmove\resources"
Tue Oct 11 2016 14:31:23.410 - psmove: CServerDriver_PSMoveService::LaunchPSMoveMonitor() monitor_psmove windows current directory: "C:\Program Files (x86)\Steam\steamapps\common\SteamVR\drivers\psmove\bin\win64\"
Tue Oct 11 2016 14:31:27.829 - Process vrmonitor (13348) has initiated a quit all
```

**I need the vrmonitor.txt log to debug this further.**

**Comment by ToppestOfDogs** [comment]
Oh sorry, I thought I put it in
[vrmonitor.txt](https://github.com/cboulay/PSMoveService/files/522653/vrmonitor.txt)

**Comment by HipsterSloth** [comment]
Well that is wierd. Right after starting it's initiating a shutdown request:

```
Tue Oct 11 2016 14:31:27.818 - [System] Transition from 'Ready' to 'Preparing to Quit...'.
Tue Oct 11 2016 14:31:27.829 - [System] Quit gracefully, power off devices and quit processes via dashboard ...
Tue Oct 11 2016 14:31:27.829 - [System] Quit gracefully: Wait for quit...
Tue Oct 11 2016 14:31:44.308 - [System] Transition from 'Preparing to Quit...' to 'Not Ready'.
Tue Oct 11 2016 14:31:44.308 - [System] System quits during startup
Tue Oct 11 2016 14:31:45.306 - [System] Transition from 'Preparing to Quit...' to 'Not Ready'.
Tue Oct 11 2016 14:31:45.306 - [System] System quits during startup
```

You mentioned you are turning on the controller after starting psmoveservice and steamvr:

> but as soon as I turned the controller on by pressing the PS button, it stopped working and did the same thing as before.

I think the PS button is mapped to the system button by default. Maybe the button press flag is getting stuck on at launch? Perhaps a held system button press terminates steamvr? 

What happens if you:
1. Shut down psmove service and steam vr
2. Kill any zombie processes
3. Start PSMoveService
4. Turn your psmove controller on
5. Launch steamvr

I usually have my PSMoveController turned on before launching SteamVR, so if there was a problem with turning on the controller post SteamVR launch I wouldn't have noticed it. I'll have to give that a test tonight when I get home.

**Comment by HipsterSloth** [comment]
Also since this sounds like this isn't in fact an issue with zombie process but rather turning on a controller with steamvr active, I'm going to re-open this as a unique issue and re-label the bug.

**Comment by ToppestOfDogs** [comment]
That's usually what I do, I'll try it again.
Okay I:
1. Shut down everything, killed all steam related processes
2. Started PSMoveService
3. Turned the controller on
4. Launched Steam VR
And then did the same again, launching RiftCat before starting the service. They both crashed the same way 
[vrserver.txt](https://github.com/cboulay/PSMoveService/files/522693/vrserver.txt)
[vrmonitor.txt](https://github.com/cboulay/PSMoveService/files/522694/vrmonitor.txt)

**Comment by HipsterSloth** [comment]
Yeah same story as before. Wierd quit signal in the vrmonitor logs. I just noticed that the vrcompositor disconnection message:

```
Tue Oct 11 2016 14:56:44.963 - Using existing HMD oculus_vridge.0.8
Tue Oct 11 2016 14:56:44.964 - New Connect message from C:\Program Files (x86)\Steam\Steam.exe (VRApplication_Overlay) 8360 
Tue Oct 11 2016 14:56:44.967 - Using existing HMD oculus_vridge.0.8
Tue Oct 11 2016 14:56:56.750 - Process vrmonitor (10604) has initiated a quit all
Tue Oct 11 2016 14:57:21.740 - IPCPipe(vrcompositor (13552)): Process disconnected.
Tue Oct 11 2016 14:57:21.740 - Process vrcompositor (13552) disconnected
```

Could you post your vrcompositor.txt logs? There might be one named "vrclient_vridge.txt"? That could be useful.

Also once you start steamvr is are there any interesting log messages that pop up in the PSMoveService log window? There should be some messages about a client connecting.

**Comment by ToppestOfDogs** [comment]
There's no vrclient_vridge.txt, but there is vrclient_PSMoveConfigTool.txt.
Here is the PSMoveService log:
```
[2016-10-11 15:21:10.793]: main - Starting PSMoveServiceConfig file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\ControllerManagerConfig.json"
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\ControllerManagerConfig.json"
[2016-10-11 15:21:10.799]: bluetooth_get_host_address - Found a bluetooth radio[2016-10-11 15:21:10.800]: bluetooth_get_host_address - Retrieved radio infoConfig file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\TrackerManagerConfig.json"
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\TrackerManagerConfig.json"
[2016-10-11 15:21:10.945]: PS3EyeTracker::open - Opening PS3EyeTracker(USB\VID_1415&PID_2000\10\0, camera_index=0)libusb: error [init_device] device '\\.\USB#VID_1EA7&PID_0908&MI_02#6&469E197&0&0002' is no longer connected!
libusb: error [init_device] device '\\.\USB#VID_18F8&PID_0F97&MI_01#6&3134A3D1&0&0001' is no longer connected!
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
ps3eye::PS3EYECam::getDevices() found 2 devices.
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\PS3EyeTrackerConfig_ps3eye_b2_p1.json"
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\PS3EyeTrackerConfig_ps3eye_b2_p1.json"
[2016-10-11 15:21:11.121]: SharedMemory::initialize() - Allocating shared memory: tracker_view_0[2016-10-11 15:21:11.252]: DeviceTypeManager::update_connected_devices - Device device_id 0 (PSEYE) opened[2016-10-11 15:21:11.253]: PS3EyeTracker::open - Opening PS3EyeTracker(USB\VID_1415&PID_2000\11\1, camera_index=1)ps3eye::PS3EYECam::getDevices() found 2 devices.
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\PS3EyeTrackerConfig_ps3eye_b2_p2.json"
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\PS3EyeTrackerConfig_ps3eye_b2_p2.json"
[2016-10-11 15:21:11.418]: SharedMemory::initialize() - Allocating shared memory: tracker_view_1[2016-10-11 15:21:11.426]: DeviceTypeManager::update_connected_devices - Device device_id 1 (PSEYE) opened[2016-10-11 15:21:13.835]: PSMoveController::open - Opening PSMoveController(\\?\hid#{00001124-0000-1000-8000-00805f9b34fb}_vid&0002054c_pid&03d5&col01#9&18c155d4&2&0000#{4d1e55b2-f16f-11cf-88cb-001111000030})[2016-10-11 15:21:13.837]: PSMoveController::open -   with serial_number: 0006f517fa60Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\00_06_f5_17_fa_60.json"
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\00_06_f5_17_fa_60.json"
[2016-10-11 15:21:13.966]: DeviceTypeManager::update_connected_devices - Device device_id 0 (PSMove) opened[2016-10-11 15:21:20.778]: ClientConnection::start - Starting client connection id 0[2016-10-11 15:21:20.781]: ClientConnection::send_connection_info - Sending connection id to client 0[2016-10-11 15:21:20.795]: ClientConnection::handle_tcp_read_request_header - Failed to read header on connection 0: An existing connection was forcibly closed by the remote host[2016-10-11 15:21:20.798]: ClientConnection::stop - Stopping client connection id 0[2016-10-11 15:21:20.800]: ClientConnection::start - Starting client connection id 1[2016-10-11 15:21:20.802]: ClientConnection::send_connection_info - Sending connection id to client 1[2016-10-11 15:22:01.377]: ClientConnection::handle_tcp_read_request_header - Failed to read header on connection 1: An existing connection was forcibly closed by the remote host[2016-10-11 15:22:01.380]: ClientConnection::stop - Stopping client connection id 1
```
I reinstalled Steam VR and installed the driver again, same thing happens.
[vrserver.txt](https://github.com/cboulay/PSMoveService/files/522764/vrserver.txt)
[vrmonitor.txt](https://github.com/cboulay/PSMoveService/files/522765/vrmonitor.txt)
[vrcompositor.txt](https://github.com/cboulay/PSMoveService/files/522766/vrcompositor.txt)
[vrclient_PSMoveConfigTool.txt](https://github.com/cboulay/PSMoveService/files/522767/vrclient_PSMoveConfigTool.txt)

**Comment by HipsterSloth** [comment]
I saw some errors about driver null appearing in the vrserver.txt

```
Tue Oct 11 2016 14:54:49.682 - error VRInitError_Init_HmdNotFound when initing driver null from C:\Program Files (x86)\Steam\steamapps\common\SteamVR\drivers\null\bin\win64\driver_null.dll.
Tue Oct 11 2016 14:54:49.682 - Unable to load driver null because of error 108. Skipping.
Tue Oct 11 2016 14:54:49.682 - Active HMD not found. Returning best error VRInitError_Init_HmdNotFound
Tue Oct 11 2016 14:54:49.682 - Refusing connect from vrcompositor (11092) because VRInitError_Init_HmdNotFound
```

Though that last line about about refusing to connect to compositor sounds bad. Can you post you steamvr.vrsettings file here? I'm curious if there is anything that stands out there.

I'm assuming that because I see the "oculus_vridge" line show up in the vrserver folder that you are using a GearVR? 

@POEDaley Do you need driver_null enabled for the gear_vr?

**Comment by ToppestOfDogs** [comment]
I'm not using a GearVR, just a generic plastic shell with lenses. I think oculus_vridge is just what riftcat calls itself. And I just copied the example script for the vrsettings.
[steamvr.zip](https://github.com/cboulay/PSMoveService/files/522818/steamvr.zip)

**Comment by POEDaley** [comment]
I am using gear_vr.  For recent testing though I have not been since I just wanted to keep the phone outside of the HMD for testing purposes.  Normally, both versions of Riftcat worked, so I didn't see any harm using either or.

Here are the requested logs:

[steamvr.zip](https://github.com/cboulay/PSMoveService/files/522824/steamvr.zip)

**Comment by HipsterSloth** [comment]
@ToppestOfDogs Your steamvr.vrsettings file looks fine as near as I can tell.

I did however just noticed you are getting a network connection error in the PSMoveService log you posted earlier (reformatted with newlines before the timestamps):

```
[2016-10-11 15:21:20.778]: ClientConnection::start - Starting client connection id 0
[2016-10-11 15:21:20.781]: ClientConnection::send_connection_info - Sending connection id to client 0
[2016-10-11 15:21:20.795]: ClientConnection::handle_tcp_read_request_header - Failed to read header on connection 0: An existing connection was forcibly closed by the remote host
[2016-10-11 15:21:20.798]: ClientConnection::stop - Stopping client connection id 0
[2016-10-11 15:21:20.800]: ClientConnection::start - Starting client connection id 1
[2016-10-11 15:21:20.802]: ClientConnection::send_connection_info - Sending connection id to client 1
[2016-10-11 15:22:01.377]: ClientConnection::handle_tcp_read_request_header - Failed to read header on connection 1: An existing connection was forcibly closed by the remote host
[2016-10-11 15:22:01.380]: ClientConnection::stop - Stopping client connection id 1
```

Did you re-run `SteamVR_ReinstallDriverWin64.bat` when you got the latest version of PSMoveService? Sorry if you already did this but I just want to make sure. I ask because the format of the network protocol can change between releases. If the old PSMove client DLL is installed in steamvr and is used with the new PSMoveService you'll get weird networking issues.

You can also verify that you have the latest driver installed by:
1) Shut down steamvr
2) Make sure: `Program Files (x86)\Steam\steamapps\common\SteamVR\drivers\psmove` exists
3) Run `SteamVR_UninstallDriverWin64.bat`
4) Make sure: `Program Files (x86)\Steam\steamapps\common\SteamVR\drivers\psmove` no longer exists
5) Run `SteamVR_ReinstallDriverWin64.bat`
6) Make sure: `Program Files (x86)\Steam\steamapps\common\SteamVR\drivers\psmove` exists again

**Comment by HipsterSloth** [comment]
@ToppestOfDogs  Also you might try out POEDaley's version of `steamvr.vrsettings`. It looks like he has driver_vridge section you don't and also has some settings removed from the steamvr section:

```
{
   "driver_vridge" : {
      "renderScale" : 160,
      "windowHeight" : 1440,
      "windowWidth" : 2560
   },
   "psmove" : {
      "circle" : "a",
      "cross" : "button_8",
      "move" : "touchpad",
      "ps" : "system",
      "rumble_suppressed" : true,
      "select" : "button_9",
      "square" : "grip",
      "start" : "button_10",
      "triangle" : "application_menu",
      "trigger" : "trigger"
   },
   "steamvr" : {
      "activateMultipleDrivers" : true,
      "mirrorViewGeometry" : "0 0 2048 1152"
   }
}
```

**Comment by ToppestOfDogs** [comment]
Alright, I did all that and changed the configuration to that one, nothing seems to have changed.
[vrserver.txt](https://github.com/cboulay/PSMoveService/files/523016/vrserver.txt)
[vrmonitor.txt](https://github.com/cboulay/PSMoveService/files/523017/vrmonitor.txt)

**Comment by HipsterSloth** [comment]
After you launch steamvr do you still see the following in the PSMoveService log window:

```
[2016-10-11 15:21:20.802]: ClientConnection::send_connection_info - Sending connection id to client 1
[2016-10-11 15:22:01.377]: ClientConnection::handle_tcp_read_request_header - Failed to read header on connection 1: An existing connection was forcibly closed by the remote host
[2016-10-11 15:22:01.380]: ClientConnection::stop - Stopping client connection id 1
```
 Or any other errors about forcibly closing connections?

**Comment by ToppestOfDogs** [comment]
No, it doesn't do anything after the controller connects

**Comment by HipsterSloth** [comment]
Well I'm out of simple things to check. I think the next best thing is for me to see if I can reproduce the issue when I get home tonight. Thanks for taking the time to try and debug this with me.

**Comment by ToppestOfDogs** [comment]
Thanks for helping

**Comment by POEDaley** [comment]
Toppest, I think the issue I'm having is similar issue to what you are experiencing.  I narrowed it down to something in the update to 5.2 that caused it for me.  Have you tested 5.0 to see if it makes a difference?

**Comment by ToppestOfDogs** [comment]
I've tried 5.1 and it worked before but it doesn't work now.

**Comment by HipsterSloth** [comment]
Here is a custom build with some additional logging that will show up in vrserver.txt

[PSMoveService-v0.9alpha5_3_1_custom.zip](https://github.com/cboulay/PSMoveService/files/523671/PSMoveService-v0.9alpha5_3_1_custom.zip)

One thing I noticed when I was running with the additional logging is that my plugin gets initialized quickly twice in a row:

```
Tue Oct 11 2016 23:37:35.603 - psmove: CServerDriver_PSMoveService::Init - called.
Tue Oct 11 2016 23:37:35.603 - psmove: CServerDriver_PSMoveService::ReconnectToPSMoveService - called.
Tue Oct 11 2016 23:37:35.603 - psmove: CServerDriver_PSMoveService::ReconnectToPSMoveService - Existing PSMoveService connection NOT active.
Tue Oct 11 2016 23:37:35.603 - psmove: CServerDriver_PSMoveService::ReconnectToPSMoveService - Starting PSMoveService connection...
Tue Oct 11 2016 23:37:35.606 - psmove: CServerDriver_PSMoveService::ReconnectToPSMoveService - Successfully connected
Tue Oct 11 2016 23:37:35.606 -   driver psmove implements interfaces IVRSettings_001 ITrackedDeviceServerDriver_004 IVRDisplayComponent_002 IVRDriverDirectModeComponent_001 IVRControllerComponent_001 IVRCameraComponent_002 IServerTrackedDeviceProvider_003 IClientTrackedDeviceProvider_004 
Tue Oct 11 2016 23:37:35.606 - Loaded server driver psmove (IServerTrackedDeviceProvider_003) from E:\Program Files (x86)\Steam\steamapps\common\SteamVR\drivers\psmove\bin\win64\driver_psmove.dll
Tue Oct 11 2016 23:37:35.607 - psmove: CServerDriver_PSMoveService::Init - called.
Tue Oct 11 2016 23:37:35.607 - psmove: CServerDriver_PSMoveService::ReconnectToPSMoveService - called.
Tue Oct 11 2016 23:37:35.607 - psmove: CServerDriver_PSMoveService::ReconnectToPSMoveService - Existing PSMoveService connection active. Shutting down...
Tue Oct 11 2016 23:37:35.613 - psmove: CServerDriver_PSMoveService::ReconnectToPSMoveService - Existing PSMoveService connection stopped.
Tue Oct 11 2016 23:37:35.613 - psmove: CServerDriver_PSMoveService::ReconnectToPSMoveService - Starting PSMoveService connection...
Tue Oct 11 2016 23:37:35.614 - psmove: CServerDriver_PSMoveService::ReconnectToPSMoveService - Successfully connected
```

I'm wondering if that leads to the connection getting stuck on the second attempt in some cases.

**Comment by HipsterSloth** [comment]
@POEDaley Oh also I tried reproducing the issue in 5.3 with vridge at home (up to this point I had been testing with my Vive setup). Unfortunately I couldn't repro the issue (game and controllers streamed correctly to the phone). I'm hoping something different shows up in your logs with the custom build.

**Comment by ToppestOfDogs** [comment]
Here's the PSMoveService log:
```
[2016-10-12 12:37:39.75]: main - Starting PSMoveServiceConfig file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\ControllerManagerConfig.json"
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\ControllerManagerConfig.json"
[2016-10-12 12:37:39.85]: bluetooth_get_host_address - Found a bluetooth radio[2016-10-12 12:37:39.86]: bluetooth_get_host_address - Retrieved radio infoConfig file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\TrackerManagerConfig.json"
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\TrackerManagerConfig.json"
[2016-10-12 12:37:39.232]: PS3EyeTracker::open - Opening PS3EyeTracker(USB\VID_1415&PID_2000\10\0, camera_index=0)libusb: error [init_device] device '\\.\USB#VID_1EA7&PID_0908&MI_02#6&469E197&0&0002' is no longer connected!
libusb: error [init_device] device '\\.\USB#VID_18F8&PID_0F97&MI_01#6&3134A3D1&0&0001' is no longer connected!
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
libusb: error [windows_get_device_list] program assertion failed - existing device should share parent
ps3eye::PS3EYECam::getDevices() found 2 devices.
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\PS3EyeTrackerConfig_ps3eye_b2_p1.json"
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\PS3EyeTrackerConfig_ps3eye_b2_p1.json"
[2016-10-12 12:37:39.407]: SharedMemory::initialize() - Allocating shared memory: tracker_view_0[2016-10-12 12:37:39.565]: DeviceTypeManager::update_connected_devices - Device device_id 0 (PSEYE) opened[2016-10-12 12:37:39.566]: PS3EyeTracker::open - Opening PS3EyeTracker(USB\VID_1415&PID_2000\11\1, camera_index=1)ps3eye::PS3EYECam::getDevices() found 2 devices.
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\PS3EyeTrackerConfig_ps3eye_b2_p2.json"
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\PS3EyeTrackerConfig_ps3eye_b2_p2.json"
[2016-10-12 12:37:39.742]: SharedMemory::initialize() - Allocating shared memory: tracker_view_1[2016-10-12 12:37:39.748]: DeviceTypeManager::update_connected_devices - Device device_id 1 (PSEYE) opened[2016-10-12 12:37:44.125]: PSMoveController::open - Opening PSMoveController(\\?\hid#{00001124-0000-1000-8000-00805f9b34fb}_vid&0002054c_pid&03d5&col01#9&18c155d4&2&0000#{4d1e55b2-f16f-11cf-88cb-001111000030})[2016-10-12 12:37:44.130]: PSMoveController::open -   with serial_number: 0006f517fa60Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\00_06_f5_17_fa_60.json"
Config file name: "C:\Users\AFlen\AppData\Roaming\PSMoveService\00_06_f5_17_fa_60.json"
[2016-10-12 12:37:44.449]: DeviceTypeManager::update_connected_devices - Device device_id 0 (PSMove) opened[2016-10-12 12:38:09.314]: ClientConnection::start - Starting client connection id 0[2016-10-12 12:38:09.316]: ClientConnection::send_connection_info - Sending connection id to client 0[2016-10-12 12:38:09.318]: ClientConnection::handle_tcp_read_request_header - Failed to read header on connection 0: An established connection was aborted by the software in your host machine[2016-10-12 12:38:09.321]: ClientConnection::stop - Stopping client connection id 0[2016-10-12 12:38:09.322]: ClientConnection::start - Starting client connection id 1[2016-10-12 12:38:09.325]: ClientConnection::send_connection_info - Sending connection id to client 1[2016-10-12 12:38:46.568]: ClientConnection::handle_tcp_read_request_header - Failed to read header on connection 1: An existing connection was forcibly closed by the remote host[2016-10-12 12:38:46.572]: ClientConnection::stop - Stopping client connection id 1
```
And the steam logs
[vrserver.txt](https://github.com/cboulay/PSMoveService/files/525119/vrserver.txt)
[vrmonitor.txt](https://github.com/cboulay/PSMoveService/files/525123/vrmonitor.txt)

**Comment by HipsterSloth** [comment]
Yeah it looks like you're getting the double init() call in your vrserver log as well

```
Wed Oct 12 2016 12:38:09.303 - psmove: CServerDriver_PSMoveService::Init - called.
Wed Oct 12 2016 12:38:09.303 - psmove: CServerDriver_PSMoveService::ReconnectToPSMoveService - called.
Wed Oct 12 2016 12:38:09.303 - psmove: CServerDriver_PSMoveService::ReconnectToPSMoveService - Existing PSMoveService connection NOT active.
Wed Oct 12 2016 12:38:09.303 - psmove: CServerDriver_PSMoveService::ReconnectToPSMoveService - Starting PSMoveService connection...
Wed Oct 12 2016 12:38:09.306 - psmove: CServerDriver_PSMoveService::ReconnectToPSMoveService - Successfully connected
Wed Oct 12 2016 12:38:09.306 -   driver psmove implements interfaces IVRSettings_001 ITrackedDeviceServerDriver_004 IVRDisplayComponent_002 IVRDriverDirectModeComponent_001 IVRControllerComponent_001 IVRCameraComponent_002 IServerTrackedDeviceProvider_003 IClientTrackedDeviceProvider_004 
Wed Oct 12 2016 12:38:09.306 - Loaded server driver psmove (IServerTrackedDeviceProvider_003) from C:\Program Files (x86)\Steam\steamapps\common\SteamVR\drivers\psmove\bin\win64\driver_psmove.dll
Wed Oct 12 2016 12:38:09.306 - psmove: CServerDriver_PSMoveService::Init - called.
```

I have another test build that ignores the second init() call from steam vr. Would you mind giving this a try:

[PSMoveService-v0.9alpha5_3_2_custom.zip](https://github.com/cboulay/PSMoveService/files/526192/PSMoveService-v0.9alpha5_3_2_custom.zip)

**Comment by ToppestOfDogs** [comment]
Same issues
[vrmonitor.txt](https://github.com/cboulay/PSMoveService/files/527976/vrmonitor.txt)
[vrserver.txt](https://github.com/cboulay/PSMoveService/files/527979/vrserver.txt)

**Comment by HipsterSloth** [comment]
I believe the original bug that this issue was opened for is now resolved in the release 5.4:
https://github.com/cboulay/PSMoveService/releases/tag/v0.9-alpha5.4
